sklearn/tree/_criterion.pyx

```diff
@@ -962,6 +963,108 @@ cdef class MSE(RegressionCriterion):
         impurity_left[0] /= self.n_outputs
         impurity_right[0] /= self.n_outputs
 
+cdef class MAE(RegressionCriterion):
+    """Mean absolute error impurity criterion
+    """
+    cdef void node_value(self, double* dest) nogil:
+        """Computes the node value of samples[start:end] into dest."""
+        with gil:
+            print "entered node_value"
+        cdef double* sample_weight = self.sample_weight
+        cdef SIZE_t* samples = self.samples
+        
+        cdef DOUBLE_t* y = self.y
+        cdef SIZE_t start = self.start
+        cdef SIZE_t end = self.end
+        cdef SIZE_t i
+        cdef SIZE_t p
+        cdef SIZE_t k
+        cdef DOUBLE_t w = 1.0
+        cdef DOUBLE_t y_ik
+
+        cdef DOUBLE_t sum_weights = 0.0
+        cdef SIZE_t median_index = 0
+        cdef DOUBLE_t Sum
+
+        y_vals = NULL
+        weights = NULL
+        y_vals = <double*> calloc(self.n_node_samples, sizeof(double))
+        weights = <double*> calloc(self.n_node_samples, sizeof(double))
+        cdef double* y_val_pointer = y_vals
+        cdef double* weight_pointer = weights
+        for k in range(self.n_outputs):
+            for p in range(start,end):
+                i = samples[p]
+
+                if sample_weight != NULL:
+                    w = sample_weight[i]
+
+                y_ik = y[i * self.y_stride + k]
+                
+                y_val_pointer[p] = y_ik
+                weight_pointer[p] = w
+
+            for p in range(start, end):
+                sum_weights += weight_pointer[p]
+                
+            Sum = sum_weights - weight_pointer[0]
+
+            while(Sum > sum_weights/2):
+                median_index +=1
+                Sum -= weight_pointer[median_index]
+
+            with gil:
+                print "calculated weighted median:"
+                print y_val_pointer[median_index]
+            dest[k] = y_val_pointer[median_index]
```

Comment:
The code runs into a `Bus Error: 10` at this line. Perhaps this is because my destination pointer as called in `node_impurity()` isn't a collection / array?